predicators/approaches/pg3_approach.py

```diff
@@ -293,6 +293,84 @@ def _get_candidate_conditions(
         return conditions
 
 
+class _DeleteConditionPG3SearchOperator(_PG3SearchOperator):
+    """An operator that removes conditions from existing LDL rules."""
+
+    def get_successors(
+            self, ldl: LiftedDecisionList) -> Iterator[LiftedDecisionList]:
+        for rule_idx, rule in enumerate(ldl.rules):
+            for destination in ("pos", "neg", "goal"):
+                if destination == "pos":
+                    dest_set = set(rule.pos_state_preconditions)
+                elif destination == "neg":
+                    dest_set = set(rule.neg_state_preconditions)
+                else:
+                    assert destination == "goal"
+                    dest_set = set(rule.goal_preconditions)
+
+                for condition in dest_set:
+                    # if the condition to be removed
+                    #    is a precondition of an nsrt, don't remove it
+                    if condition in rule.nsrt.preconditions:
+                        continue
+
+                    #rcreate new preconditions
+                    new_pos = set(rule.pos_state_preconditions)
+                    new_neg = set(rule.neg_state_preconditions)
+                    new_goal = set(rule.goal_preconditions)
+
+                    # remove the focus predicate
+                    if destination == "pos":
+                        new_pos.remove(condition)
+                    elif destination == "neg":
+                        new_neg.remove(condition)
+                    else:
+                        assert destination == "goal"
+                        new_goal.remove(condition)
+
+                    # check that the parameters of the removed
```

Comment:
this looks reasonable, but it might be cleaner to just totally reconstruct the parameters of the rule like this: https://github.com/Learning-and-Intelligent-Systems/predicators/blob/master/predicators/approaches/initialized_pg3_approach.py#L143

Reply:
(in fact, maybe we should just write a helper function to do this and use it everywhere)